lib/AST/Decl.cpp

```diff
@@ -2450,6 +2450,10 @@ DestructorDecl *ClassDecl::getDestructor() {
 }
 
 bool ClassDecl::inheritsSuperclassInitializers(LazyResolver *resolver) {
+  // Get a resolver from the ASTContext if we don't have one already.
+  if (resolver == nullptr)
+    resolver = getASTContext().getLazyResolver();
```

Comment:
Please don't do this. Either we should convert everything over to getting it from the ASTContext, or we should respect someone passing null.